File: app-service-comments/src/main/java/com/github/arhor/aws/graphql/federation/comments/data/repository/PostRepository.java
package com.github.arhor.aws.graphql.federation.comments.data.repository;

import com.github.arhor.aws.graphql.federation.comments.data.entity.PostEntity;
import org.springframework.data.repository.ListCrudRepository;

import java.util.UUID;

public interface PostRepository extends ListCrudRepository<PostEntity, UUID> {
}

File: dgs-posts/src/main/kotlin/com/github/arhor/dgs/posts/data/repository/PostRepository.kt
package com.github.arhor.dgs.posts.data.repository

import com.github.arhor.dgs.posts.data.entity.PostEntity
import com.github.arhor.dgs.posts.data.entity.projection.PostProjection
import org.springframework.data.jdbc.repository.query.Query
import org.springframework.data.repository.ListCrudRepository
import java.util.stream.Stream

interface PostRepository : ListCrudRepository<PostEntity, Long> {

    @Query(name = "PostProjection.findAll")
    fun findAll(limit: Long, offset: Long): List<PostProjection>

    fun findAllByUserId(userId: Long): Stream<PostEntity>

    @Query(name = "PostProjection.findAllByUserIdIn")
    fun findAllByUserIdIn(userIds: Collection<Long>): List<PostProjection>
}
